oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/ElasticsearchIndexDefinition.java

```diff
@@ -18,11 +18,100 @@
  */
 package org.apache.jackrabbit.oak.plugins.index.elasticsearch;
 
+import org.apache.jackrabbit.oak.commons.PathUtils;
 import org.apache.jackrabbit.oak.plugins.index.search.IndexDefinition;
 import org.apache.jackrabbit.oak.spi.state.NodeState;
 
-public class ElasticsearchIndexDefinition extends IndexDefinition {
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static org.apache.jackrabbit.oak.plugins.index.search.util.ConfigUtil.getOptionalValue;
+import static org.elasticsearch.common.Strings.INVALID_FILENAME_CHARS;
+
+public class ElasticsearchIndexDefinition extends IndexDefinition implements ElasticsearchIndexConstants {
+
+    private static final int MAX_NAME_LENGTH = 255;
+
+    private static final String INVALID_CHARS_REGEX = Pattern.quote(INVALID_FILENAME_CHARS
+            .stream()
+            .map(Object::toString)
+            .collect(Collectors.joining("")));
+
+    private final String elasticsearchIndexName;
+
+    public final int bulkActions;
+    public final long bulkSizeBytes;
+    public final long bulkFlushIntervalMs;
+    public final int bulkRetries;
+    public final long bulkRetriesBackoff;
+
     public ElasticsearchIndexDefinition(NodeState root, NodeState defn, String indexPath) {
         super(root, getIndexDefinitionState(defn), determineIndexFormatVersion(defn), determineUniqueId(defn), indexPath);
+        this.elasticsearchIndexName = getRemoteIndexName();
+
+        this.bulkActions = getOptionalValue(defn, BULK_ACTIONS, BULK_ACTIONS_DEFAULT);
+        this.bulkSizeBytes = getOptionalValue(defn, BULK_SIZE_BYTES, BULK_SIZE_BYTES_DEFAULT);
+        this.bulkFlushIntervalMs = getOptionalValue(defn, BULK_FLUSH_INTERVAL_MS, BULK_FLUSH_INTERVAL_MS_DEFAULT);
+        this.bulkRetries = getOptionalValue(defn, BULK_RETRIES, BULK_RETRIES_DEFAULT);
+        this.bulkRetriesBackoff = getOptionalValue(defn, BULK_RETRIES_BACKOFF, BULK_RETRIES_BACKOFF_DEFAULT);
+    }
+
+    public String getElasticsearchIndexName() {
```

Comment:
Can we override the base class method `getIndexName` instead of creating a new method?

Reply:
there is no 1 to 1 mapping between the OAK index name and index name in ES. In the future, for scalability reasons, we could also have multiple indexes in ES related to an OAK index definition. We cannot override `getIndexName`. I will add some Javadoc on `getElasticsearchIndexName`. Maybe a different name would help? Something like `getRemoteIndexName`?

Reply:
Yes. `getRemoteIndexName` would clear the misunderstanding :)